gateway/handlers/reader.go

```diff
@@ -11,31 +11,43 @@ import (
 
 	"strings"
 
-	"github.com/openfaas/faas/gateway/metrics"
-	"github.com/openfaas/faas/gateway/requests"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/swarm"
 	"github.com/docker/docker/client"
+	"github.com/openfaas/faas/gateway/metrics"
+	"github.com/openfaas/faas/gateway/requests"
 )
 
 // MakeFunctionReader gives a summary of Function structs with Docker service stats overlaid with Prometheus counters.
 func MakeFunctionReader(metricsOptions metrics.MetricOptions, c *client.Client) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 
+		verbose := false
+		verboseMode := r.URL.Query().Get("v")
+		if len(verboseMode) != 0 {
+			verbose = true
+		}
 		serviceFilter := filters.NewArgs()
 
 		options := types.ServiceListOptions{
 			Filters: serviceFilter,
 		}
 
-		services, err := c.ServiceList(context.Background(), options)
+		ctx := context.Background()
+		services, err := c.ServiceList(ctx, options)
 		if err != nil {
 			fmt.Println(err)
 		}
 
 		// TODO: Filter only "faas" functions (via metadata?)
 		var functions []requests.Function
 
+		var running map[string]int
+		if verbose {
```

Comment:
At this point I wonder if we need to add GET /system/function/<function_name> that does the verbose query if we want to look up the status of only one function from the UI or similar.

Reply:
This would be useful, you'll see in my PR to @austinfrey's Node library that I had to filter the returned response from `/system/functions` to find the relevant function. I think the more RESTful the API, the better 👍

https://github.com/developius/node-openfaas/blob/e569e50b61b7190b6ec8806bb96eda1343e28437/openfaas/index.js#L23-L41